src/MoBi.Core/Domain/Model/MoBiSimulation.cs

```diff
@@ -73,9 +86,38 @@ public void RemoveAllChartTemplates()
 
       public bool IsCreatedBy(IBuildingBlock templateBuildingBlock)
       {
-         //TODO SIMULATION_CONFIGURATION
+         // We can consider the building block in-use if it belongs to a module that is in use.
+         if (templateBuildingBlock.Module != null)
+            return usesModuleBuildingBlock(templateBuildingBlock);
+
+         // Simple name match for building blocks that do not belong to a module
+         switch (templateBuildingBlock)
+         {
+            case IndividualBuildingBlock individualBuildingBlock:
+               return Equals(Configuration.Individual?.Name, individualBuildingBlock.Name);
+            case ExpressionProfileBuildingBlock expressionProfileBuildingBlock:
+               return Configuration.ExpressionProfiles.Any(x => Equals(x.Name, expressionProfileBuildingBlock.Name));
+         }
+
          return false;
-         // return MoBiBuildConfiguration.BuildingInfoForTemplate(templateBuildingBlock) != null;
+      }
+
+      private bool usesModuleBuildingBlock(IBuildingBlock templateBuildingBlock)
+      {
+         return BuildingBlocks.Any(buildingBlock => buildingBlock.Module.IsNamed(templateBuildingBlock.Module.Name) && buildingBlock.IsTemplateMatchFor(templateBuildingBlock));
+      }
+
+      public IReadOnlyList<IBuildingBlock> BuildingBlocks
+      {
+         get
+         {
+            var buildingBlocks = Configuration.ModuleConfigurations.SelectMany(moduleConfiguration => moduleConfiguration.Module.BuildingBlocks).Concat(Configuration.ExpressionProfiles).ToList();
```

Comment:
maybe we should cache some of this shit (e.g. all building blocs used in a simulation)(. This never changes once the simulation is created and could be saved in the config or simulations. THis is repeated just too often I find

Reply:
I'm with you on making an accessor that gives me all the modules like ```simulation.Modules {get}```. But caching them? I don't think it takes too long to calculate the modules for a simulation, and from that all the building blocks for the modules.

This way it won't be repeated.

Reply:
not the modules but the building blocks. THis never changes once it's created. Probalby not required unless this access is done over all parameters withing a loop etc.. . If we cached, we could go back to get get method and not a function